NSRetailPOS/NSRetailPOS/NSRetailPOS/Operations/Item/frmItemCodeList.cs

```diff
@@ -0,0 +1,364 @@
+
+                    if (Convert.ToBoolean(itemObj.IsNewToggleSwitched))
+                    {
+                        btnNew_Click(sender, e);
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                XtraMessageBox.Show(ex.Message);
+            }
+        }
+
+        private void btnDelete_Click(object sender, EventArgs e)
+        {
+            try
+            {
+                if (gvItemList.FocusedRowHandle >= 0 &&
+                    XtraMessageBox.Show($"Are you sure you want to delete the Item Code : {gvItemList.GetFocusedRowCellValue("ITEMCODE")}?"
+                    , "Confirm", MessageBoxButtons.YesNo, MessageBoxIcon.Question) == DialogResult.Yes)
+                {
+                    new ItemCodeRepository().DeleteItemCode(gvItemList.GetFocusedRowCellValue("ITEMCODEID"), Utility.loginInfo.UserID);
+                    XtraMessageBox.Show($"Delete completed successfully", "Info", MessageBoxButtons.OK, MessageBoxIcon.Information);
+                }
+            }
+            catch (Exception ex)
+            {
+                XtraMessageBox.Show(ex.Message);
+            }
+        }
+
+        private void btnVisualize_Click(object sender, EventArgs e)
+        {
+            try
+            {
+                if (gvItemList.FocusedRowHandle >= 0)
+                {
+                    new frmItemVisualize(gvItemList.GetFocusedRowCellValue("ITEMID")).ShowDialog();
+                }
+            }
+            catch (Exception ex)
+            {
+                XtraMessageBox.Show(ex.Message);
+            }
+        }
+
+
+        #endregion
+
+        protected override bool ProcessDialogKey(Keys keyData)
+        {
+            if (Form.ModifierKeys == Keys.None && keyData == Keys.Escape)
+            {
+                this.Close();
+                return true;
+            }
+            return base.ProcessDialogKey(keyData);
+        }
+
+        private void frmItemList_Load(object sender, EventArgs e)
+        {
+            try
+            {
+                EnableDisableControls(false);
+                dtItemCodes = Utility.GetItemCodeListFiltered();
+                dtItems = Utility.GetItemSKUList();
+                gcItemList.DataSource = dtItemCodes;
+                ((frmOpetations)this.MdiParent).RefreshBaseLineData += FrmItemCodeList_RefreshBaseLineData;
+            }
+            catch (Exception ex)
+            {
+                XtraMessageBox.Show(ex.Message);
+            }
+        }
+
+        private void FrmItemCodeList_RefreshBaseLineData(object sender, EventArgs e)
+        {
+            dtItemCodes = Utility.GetItemCodeListFiltered();
+            dtItems = Utility.GetItemSKUList();
+            gcItemList.DataSource = dtItemCodes;
+        }
+
+        private void gcItemList_DoubleClick(object sender, EventArgs e)
+        {
+            //btnEdit_Click(null, null);
```

Comment:
not needed?

Reply:
Yes, it is not required as we are not giving provision to edit item in POS